hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -101,29 +100,23 @@ protected String getSQLUpdateRowString() {
 		}
 
 		ReactiveConnection reactiveConnection = getReactiveConnection( session );
-
 		collection.preInsert( this );
-
-		final AtomicInteger index = new AtomicInteger( 1 );
-		return CompletionStages.total(
-				collection.entries( this ),
-				entry -> {
-					CompletionStage<Integer> st = reactiveConnection.update(
-							getSQLInsertRowString(),
-							insertRowsParamValues( entry, index, collection, id, session ) );
-					collection.afterRowInsert( this, entry, index.getAndIncrement() );
-					return st;
-				}
-		)
-				.thenAccept( total -> LOG.debugf( "Done inserting rows: %s inserted", total ) );
+		Iterator<?> entries = collection.entries(this );
+		return total(
+				entries,
+				(entry, index) -> reactiveConnection.update(
+						getSQLInsertRowString(),
+						insertRowsParamValues( entry, index+1, collection, id, session )
+				)
+				//TODO: compose() a reactive version of collection.afterRowInsert()
+		).thenAccept( total -> LOG.debugf( "Done inserting rows: %s inserted", total ) );
```

Comment:
@blafond careful with stuff like this. Calling `afterRowInsert()` after calling something which returns a `CompletionStage` almost certainly doesn't do what you intend it to do.

(Apologies for not catching this when reviewing the PR.)

Reply:
noted. thx